scripts/ansible/VM/create.yml

```diff
@@ -0,0 +1,165 @@
+      set_fact:
+        image_dict: "{{ images_list.resource.images |
+                        items2dict(key_name='name', value_name='id') }}"
+
+    - name: Check for existing VSI
+      ibm_is_instance_info:
+        name: "{{ vsi_name }}"
+      failed_when:
+        - vsi.rc != 0
+        - '"No Instance found" not in vsi.stderr'
+      register: vsi
+
+    - name: Configure VSI
+      ibm_is_instance:
+        name: "{{ vsi_name }}"
+        state: available
+        id: "{{ vsi.resource.id | default(omit) }}"
+        vpc: "{{ vpc.resource.id }}"
+        profile: "{{ vsi_profile }}"
+        image: "{{ image_dict[vsi_image] }}"
+        keys:
+         - "{{ ssh_key.resource.id }}"
+        primary_network_interface:
+         - subnet: "{{ subnet.resource.id }}"
+        zone: "{{ zone }}"
+      register: vsi
+
+    - name: Check for existing Floating IP
+      ibm_is_floating_ip_info:
+        name: "{{ fip_name }}"
+      failed_when:
+        - fip.rc != 0
+        - '"No floatingIP found" not in fip.stderr'
+      register: fip
+
+    - name: Configure Floating IP Address
+      ibm_is_floating_ip:
+        name: "{{ fip_name }}"
+        state: available
+        id: "{{ fip.resource.id | default(omit) }}"
+        target: "{{ vsi.resource.primary_network_interface[0]['id'] }}"
+      register: fip
+
+    - name: Print Floating IP Address
+      debug:
+        msg: "{{ vsi_name }} IP Address: {{ fip.resource.address }}"
+
+    - name: Configure Security Group Rule to open SSH on the VSI
+      ibm_is_security_group_rule:
+        state: available
+        group: "{{ vpc.resource.default_security_group }}"
+        direction: inbound
+        remote: 0.0.0.0/0
+        tcp:
+          - port_max: 22
+            port_min: 22
+
+    - name: Add VSI to Ansible inventory
+      add_host:
+        name: "{{ fip.resource.address }}"
+        ansible_user: root
+        groups: new_vsi
+        ansible_ssh_extra_args: -o StrictHostKeyChecking=no -i "{{ ssh_login_key }}" 
+
+- name: Check Ansible connection to new DEMO VSI
+  hosts: new_vsi
+  gather_facts: False
+  tasks:
+    - name: Wait for VSI to become reachable over SSH
+      wait_for_connection:
+
+- name: Check Ansible connection to new DEMO VSI
+  hosts: new_vsi
+  tasks:
+    - name: Collect OS information
+      command: cat /etc/os-release
+      register: os_info
+
+    - name: Print OS information
+      debug:
+        var: os_info.stdout_lines
+
+# above this line all the steps are working 
+# next steps are for installation of podman and other similar configurations
+
+- name: copy installation scirpt
+  hosts: new_vsi
+  tasks:
+    - name: copying file with playbook
+      become: true 
+      copy:
+        src: ~/go/src/github.com/redhat-developer/odo/scripts/ansible/VM/install_dependency.sh
+        dest: ~/      
+        mode: 0644
+    - name: install podman and golang
+      command: bash ~/install_dependency.sh
+      register: output
```

Comment:
I would prefer to use Ansible directives to install necessary packages, instead of a script. The Ansible directives are generally compatible with different Linux distributions, which can make life much easier whzn we need to change distribution